File: src/main/java/eterea/core/service/model/dto/TransferenciaWrapperDto.java
package eterea.core.service.model.dto;

import eterea.core.service.hexagonal.contable.cuentamovimiento.domain.model.CuentaMovimiento;
import eterea.core.service.hexagonal.tesoreria.valormovimiento.domain.model.ValorMovimiento;
import eterea.core.service.tool.Jsonifyable;
import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TransferenciaWrapperDto implements Jsonifyable {

    private TransferenciaDto transferencia;
    private List<ValorMovimiento> valorMovimientos;
    private List<CuentaMovimiento> cuentaMovimientos;

}

File: src/main/java/eterea/core/service/kotlin/model/dto/TransferenciaWrapperDto.kt
package eterea.core.service.kotlin.model.dto

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.json.JsonMapper
import eterea.core.service.kotlin.model.CuentaMovimiento
import eterea.core.service.kotlin.model.ValorMovimiento

data class TransferenciaWrapperDto(

    var transferencia: TransferenciaDto? = null,
    var valorMovimientos: List<ValorMovimiento?>? = null,
    var cuentaMovimientos: List<CuentaMovimiento?>? = null

) {

    fun jsonify(): String {
        try {
            return JsonMapper
                .builder()
                .findAndAddModules()
                .build()
                .writerWithDefaultPrettyPrinter()
                .writeValueAsString(this)
        } catch (e : JsonProcessingException) {
            return "jsonify error -> ${e.message}"
        }
    }

    class Builder {
        private var transferencia: TransferenciaDto? = null
        private var valorMovimientos: List<ValorMovimiento?>? = null
        private var cuentaMovimientos: List<CuentaMovimiento?>? = null

        fun transferencia(transferencia: TransferenciaDto?) = apply { this.transferencia = transferencia }
        fun valorMovimientos(valorMovimientos: List<ValorMovimiento?>?) = apply { this.valorMovimientos = valorMovimientos }
        fun cuentaMovimientos(cuentaMovimientos: List<CuentaMovimiento?>?) = apply { this.cuentaMovimientos = cuentaMovimientos }

        fun build() = TransferenciaWrapperDto(
            transferencia,
            valorMovimientos,
            cuentaMovimientos
        )
    }

    companion object {
        fun builder() = Builder()
    }

}
